examples/htool.c

```diff
@@ -251,6 +251,413 @@ static int do_target_reset(uint32_t reset_option) {
+    case 9:
+      return "Invalid Hash";
+    default:
+      return "(unknown payload_validation_failure_reason)";
+  }
+}
+
+const char* ImageTypeString(uint8_t type) {
+  switch (type) {
+    case 0:
+      return "Dev";
+    case 1:
+      return "Prod";
+    case 2:
+      return "Breakout";
+    case 3:
+      return "Test";
+    case 4:
+      return "UnsignedIntegrity";
+    case 255:
+      return "Unspecified";
+    default:
+      return "(unknown image_type)";
+  }
+}
+
+static int do_payload_status() {
+  struct libhoth_usb_device* dev = htool_libhoth_usb_device();
+  if (!dev) {
+    return -1;
+  }
+
+  uint8_t buffer[sizeof(struct payload_status_response_header) +
+                 2 * sizeof(struct payload_region_state)];
+  size_t rlen = 0;
+  int ret = htool_exec_hostcmd(
+      dev, EC_CMD_BOARD_SPECIFIC_BASE + EC_PRV_CMD_HOTH_PAYLOAD_STATUS, 0,
+      NULL, 0, &buffer, sizeof(buffer), &rlen);
+  printf("[do_payload_status] ret=%d, rlen=%d\n", ret, rlen);
+  if (ret != 0) {
+    printf("Error code: %d\n", ret);
+    return -1;
+  }
+
+  struct payload_status_response_header* ppsr =
+      (struct payload_status_response_header*)(buffer);
+  printf("version       : %u\n", ppsr->version);
+  printf("lockdown_state: %s (%u)\n",
+         SpsEepromLockdownStatusString(ppsr->lockdown_state),
+         ppsr->lockdown_state);
+  printf("active_half   : %c\n", ppsr->active_half ? 'B' : 'A');
+  printf("region_count  : %u\n", ppsr->region_count);
+
+  size_t expected_rlen =
+      sizeof(struct payload_status_response_header) +
+      ppsr->region_count * sizeof(struct payload_region_state);
+  if (rlen != expected_rlen) {
+    printf("rlen is %zu while expected rlen is %zu, region_count is %u.\n",
+           rlen, expected_rlen, ppsr->region_count);
+    return -1;
+  }
+
+  for (int region_index = 0; region_index < ppsr->region_count;
+       region_index++) {
+    int offset = sizeof(struct payload_status_response_header) +
+        region_index * sizeof(struct payload_region_state);
+    const struct payload_region_state* rs =
+        (struct payload_region_state*)(&buffer[offset]);
+    printf("Region %c:\n", region_index == 0 ? 'A' : 'B');
+    printf("  validation_state: %s (%u)\n",
+           PayloadValidationStateString(rs->validation_state),
+           rs->validation_state);
+    printf("  failure_reason: %s (%u)\n",
+           PayloadValidationFailureReasonString(rs->failure_reason),
+           rs->failure_reason);
+    printf("  image_type: %s (%u)\n", ImageTypeString(rs->image_type),
+           rs->image_type);
+    printf("  key_index:  %u\n", rs->key_index);
+    printf("  image_family: (0x%08x)\n", rs->image_family);
+    printf("  version: %u.%u.%u.%u\n", rs->version_major, rs->version_minor,
+           rs->version_point, rs->version_subpoint);
+    printf("  descriptor_offset: 0x%08x\n", rs->descriptor_offset);
+  }
+  return 0;
+}
+
+static int command_payload_version(const struct htool_invocation* inv) {
```

Comment:
It is removed, since the "payload version" is just a subset of "payload status". But it doesn't matter that much because all payload-related code is removed from this change for now and will be created in a separate change.